wfapi/error.go

```diff
@@ -112,39 +110,38 @@ func ErrorWorkspace(wsPath string, cause error) error {
 //
 //    - warpforge-error-executor-failed --
 func ErrorExecutorFailed(executorEngineName string, cause error) error {
-	result := serum.Errorf(ECodeExecutorFailed,
-		"executor engine failed: the %s engine reported error: %w", executorEngineName, cause)
-	addDetails(result, [][2]string{
-		{"engineName", executorEngineName},
+	return serum.Error(ECodeExecutorFailed, serum.WithCause(cause),
+		serum.WithMessageTemplate("the {{engineName|q}} engine reported error"),
+		serum.WithDetail("engineName", executorEngineName),
 		// ideally we'd have more details here, but honestly, our executors don't give us much clarity most of the time, so... we'll see.
-	})
-	return result
+	)
 }
 
+// DEPRECATED: This is just adds a degenerate repetition of the error code
 // ErrorIo wraps generic I/O errors from the Go stdlib
 //
 // Errors:
 //
 //    - warpforge-error-io --
 func ErrorIo(context string, path string, cause error) error {
-	result := serum.Errorf(ECodeIo,
-		"io error: %s: %w", context, cause)
-	addDetails(result, [][2]string{{"context", context}, {"path", path}})
-	return result
+	return serum.Error(ECodeIo, serum.WithCause(cause),
+		serum.WithMessageTemplate("io error: {{context}}"),
+		serum.WithDetail("context", context),
+		serum.WithDetail("path", path),
+	)
 }
 
+// DEPRECATED: This is just adds a degenerate repetition of the error code
```

Comment:
I think this is not accurate; I'm pretty sure this is most usually being used to wrap errors from the ipld libraries (which aren't serum'ized).

(It's maybe not the most communicative either -- ideally it would say something about if it was during serialization, or deserialization, or struct mapping, or schema validation, or so on.  But I think this currently balks at that because it would be difficult to disambiguate some of these without either upstream patches, or building an awful lot of facades.)

Reply:
My point here is that It would be _just as valuable_ to wrap errors with `serum.Error(ECodeSerialization, serum.WithCause(cause), serum.WithMessageLiteral(context))`. And I don't feel that this meets the threshold for creating a helper function.

Reply:
The deprecation comment is pretty low-signal. These are more descriptive now.